packages/checkout/widgets-lib/src/lib/hooks/useBalance.ts

```diff
@@ -0,0 +1,78 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Checkout, GetBalanceResult } from '@imtbl/checkout-sdk';
+import { Web3Provider } from '@ethersproject/providers';
+import { getTokenBalances } from '../../widgets/wallet/functions/tokenBalances';
+import { DEFAULT_BALANCE_RETRY_POLICY } from '../constants';
+import { useInterval } from './useInterval';
+
+const REFRESH_BALANCE_INTERVAL_MS = 10000;
+
+export interface UseBalanceParams {
+  checkout: Checkout | undefined;
+  provider: Web3Provider | undefined,
+  refreshCallback: (balance: GetBalanceResult[]) => void;
+  errorCallback: (error: Error) => void;
+}
+
+export const useBalance = ({
+  checkout,
+  provider,
+  refreshCallback,
+  errorCallback,
+}: UseBalanceParams) => {
+  const [balancesLoading, setBalancesLoading] = useState(true);
+
+  const refreshBalances = useCallback(async (
+    // If silent is true, the balances will not be set to loading
+    silent: boolean = false,
+  ) => {
+    if (!checkout || !provider) return;
+    try {
+      const network = await checkout.getNetworkInfo({
+        provider,
+      });
+
+      /* If the provider's network is not supported, return out of this and let the
+      connect loader handle the switch network functionality */
+      if (!network.isSupported) {
+        return;
+      }
```

Comment:
I reckon this logic is a out-dated and unnecessary -- @ZacharyCouchman might have more to say about this

Reply:
As above, provider window events will be hooked into and this refresh will most likely always be bailed out before having to check the supported underlying chain. It's just all upstream so we happy to make that call it will always be handled elsewhere?

Reply:
If unsupported network is true and this exit is removed it looks like it can still call the sdk but the sdk itself also checks the chain is supported before making blockscout calls. The provider events set the widget to the switch network screen but I can see there are still calls to the sdk unnecessarily when this check is removed